treeherder/webapp/api/utils.py

```diff
@@ -54,6 +55,22 @@ def get_repository(name):
     if name in REPO_GROUPS:
         param = REPO_GROUPS[name]
     else:
+        try:
```

Comment:
@ghickman I had a comment that disappeared when I pushed a minor fix early yesterday... 

I had a question as to whether it'd be better to put these 404's in the QueryParamsSerializer. I know a non-existent repo or bug id is not technically malformed data, but the downside to putting it here is that if a user has more than one query string param that's incorrect, the serializer will return the field error first (say startday is missing and bug id is non-existent, only the startday error will be returned in the response).  If I move these to the serializer and raise a ValidationError then all errors will return from the serializer. Thoughts? 

ex: localhost:8000/api/failuresbybug/?startday=&endday=2018-02-16&tree=mozilla-inbound&bug=122222222

Reply:
That sounds like a good idea.  Giving all the "most correct" error feedback is good. :)